tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,264 @@
+    "# security auditors, add this to this file:",
+    "#",
+    "#   apache::trace_enable: Off",
+    "#",
+    "# Consult the full module documentation on http://forge.puppetlabs.com,",
+    "# or the actual puppet classes themselves, to discover options to configure.",
+    "#",
+    "# Do note, setting some values may have unintended consequences that affect the",
+    "# performance or functionality of the application. Consider the impact of your",
+    "# changes before applying them, and test them in a non-production environment",
+    "# first.",
+    "#",
+    "# Here are some examples of how you tune the Apache options if needed:",
+    "#",
+    "# apache::mod::prefork::startservers: 8",
+    "# apache::mod::prefork::minspareservers: 5",
+    "# apache::mod::prefork::maxspareservers: 20",
+    "# apache::mod::prefork::serverlimit: 256",
+    "# apache::mod::prefork::maxclients: 256",
+    "# apache::mod::prefork::maxrequestsperchild: 4000",
+    "# Here are some examples of how you tune the PostgreSQL options if needed:",
+    "#",
+    "# postgresql::server::config_entries:",
+    "#   max_connections: 600",
+    "#   shared_buffers: 1024MB",
+]
+
+MEDIUM_TUNING_DATA = [
+    "apache::mod::passenger::passenger_max_pool_size: 30",
+    "apache::mod::passenger::passenger_max_request_queue_size: 1000",
+    "apache::mod::passenger::passenger_max_requests: 1000",
+    " ",
+    "apache::mod::prefork::serverlimit: 1024",
+    "apache::mod::prefork::maxclients: 1024",
+    "apache::mod::prefork::maxrequestsperchild: 4000",
+    " ",
+    "qpid::open_file_limit: 65536",
+    "qpid::router::open_file_limit: 150100",
+    " ",
+    "postgresql::server::config_entries:",
+    "  max_connections: 1000",
+    "  shared_buffers: 4GB",
+    "  work_mem: 4MB",
+    "  checkpoint_segments: 32",
+    "  checkpoint_completion_target: 0.9",
+    "  effective_cache_size: 16GB",
+    "  autovacuum_vacuum_cost_limit: 2000",
+    "  log_min_duration_statement: 500 "
+]
+
+MMPV1_MONGODB = [
+    "# Added by foreman-installer during upgrade, run the "
+    "installer with --upgrade-mongo-storage to upgrade to WiredTiger.",
+    "mongodb::server::storage_engine: 'mmapv1'"
+]
+
+WIREDTIGER_MONGODB = [
+    "# Do not remove",
+    "mongodb::server::storage_engine: 'wiredTiger'"
+]
+
+
+@destructive
+class ScenarioPerformanceTuning(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test
+    Performance Tuning utility
+
+    Test Steps:
+
+    1. Before Satellite upgrade, we will apply the medium tune size.
+    2. Once the size get apply we will wait for post upgrade test case to check
+    whether the size is same or not.
+    3. In post upgrade we will check whether satellite installer was able to change
+    the size or not
+
+    :expectedresults: The performance parameter should not be changed after upgrade.
+    """
+    @staticmethod
+    def _custom_hiera_file_creation(mongodb_type, tune_size="default"):
```

Comment:
Try to use verbs instead of nouns in function names, as functions do something: `_create_custom_hiera_file`.